### DISCUSSION TITLE
delayed response from audio_state_changed.

### DISCUSSION THREAD

**Discussion by leonardokucarz** [pschatzmann/ESP32-A2DP#443] [Q&A] [answered]
Hi, i'm building a project with lcd using lvgl and your librarie, and i'm trying to solve a small question, when i made the "play" command on my smartphone the response don´t have a visible delay but when i made the "pause" command it takes a while to answer.

it's the part of code where i implement this function.

```
void audio_state_changed(esp_a2d_audio_state_t state, void *ptr){
  Serial.println(a2dp_sink.to_str(state));
  Serial.printf(" %x\n", state);
  audioState = state;
  switch (audioState) {
    case 0:
      lv_obj_clear_state(ui_Panel1, LV_STATE_USER_2);
      lv_obj_add_state(ui_Panel1, LV_STATE_USER_1);
      break;

    case 2:
      lv_obj_clear_state(ui_Panel1, LV_STATE_USER_1);
      lv_obj_add_state(ui_Panel1, LV_STATE_USER_2);
      break;

  }
}
```

Here is a video showing this delay happend.

https://github.com/pschatzmann/ESP32-A2DP/assets/85085056/11710809-91ca-47e9-aab5-c14481def1f0

any help would be appreciated.  :)

**Comment by pschatzmann** [comment, marked as answer]
I can't really see any bug here. 
Why don't you just set the log level to debug to see what's going on in detail ?

**Comment by hamedhemmati** [comment]
I see the same behavior. `a2dp_sink.get_audio_state()` reports 2 right away but reporting 0 is delayed by about 2 seconds after pause has been pressed.

**Comment by pschatzmann** [comment]
Yes, but isn't this quite logical: When the task queue is empty (at the start), you will get the result immediately and when the queue is filled with audio data (while playing) it takes it's time to arrive. 

You could try to decrease the queue size, but I don't think that this is a good idea...

**Reply by hamedhemmati** [reply]
Your logic makes sense. In case of what @leonardokucarz is trying to accomplish, it makes it look laggy and slow. In my case, I have servo motors that need to adjust their speed. I am wondering if there is another way to get an immediate response since the internal ES32 board's LED stops blinking right away.

**Comment by KickingAnimal** [comment]
I have experienced the same when connected from a Samsung phone, when connected from a Windows 10 laptop the response was much quicker (about as i would excpect).
is there something that windows does fundametally different in handling bluetooth?
I also noticed that in some other ways the respose would be near instant, when closing the app or start a call.

might be a lack of understanding the library that is the problem
